client/src/scene/SpectatorUI.ts

```diff
@@ -87,10 +87,13 @@ export class SpectatorUI {
      * Destroys all GameObjects belonging to this class.
      */
     public destroy() {
+        this.removeListeners();
+        this.removeTimedEvent();
         this.header.destroy();
         this.countdown.destroy();
         this.alreadyVoted.destroy();
         this.buttons.forEach((button) => button.destroy());
+        this.countdownTimer = undefined;
```

Comment:
Could you explain why this is needed?

Reply:
I wasn't sure if clearing the interval was enough for the GC to delete the references. I threw it in to be safe but if the clearInterval is enough then this can be removed

Reply:
Not that I'm an expert but iirc JS's garbage collector operates on "reachability" and future accesses. If the ApectatorUI entity is no longer reachable (like would be with the `this.spectator = null;` suggestion I made above) or accessed, neither are its members, so they will be cleaned up as well.

Note that the `this.spectator = null;` suggestion is not explicitly needed either, but its a cheap code maintainability practice to prevent future issues where someone is trying to use `this.spectator` but it's in a "still-exists but all its game objects have been destroyed" state. Setting it to null explicitly requires the creation of a new entity.

Reply:
That makes sense. I set the spectator variable to be `undefined` since the `?` character doesn't allow for `null` (though it can be added as an additional type if we wanted).

Setting it to undefined should still be enough though correct?

Reply:
ah right, yeah. Should be fine.